hloc/extractors/dirnet.py

```diff
@@ -0,0 +1,74 @@
+import sys
+from pathlib import Path
+import subprocess
+import logging
+import torch
+
+from zipfile import ZipFile
+import os
+
+from ..utils.base_model import BaseModel
+
+dirnet_path = Path(__file__).parent / '../../third_party/deep-image-retrieval'
+
+sys.path.append(str(dirnet_path))
+
+from dirtorch import nets as nets
+from dirtorch.utils import common
+
+
+def load_model(path, iscuda):
+    checkpoint = common.load_checkpoint(path, iscuda)
+    net = nets.create_model(pretrained="", **checkpoint['model_options'])
+    net = common.switch_model_to_cuda(net, iscuda, checkpoint)
+    net.load_state_dict(checkpoint['state_dict'])
+    net.preprocess = checkpoint.get('preprocess', net.preprocess)
+    if 'pca' in checkpoint:
+        net.pca = checkpoint.get('pca')
+    return net
+
+
+class DIRnet(BaseModel):
+    default_conf = {
+        'model_name': 'Resnet-101-AP-GeM.pt',
+        'checkpoint': dirnet_path / 'dirtorch/data/Resnet-101-AP-GeM.pt',
+        'whiten': 'Landmarks_clean',
+        'whitenp': 0.25,
+        'whitenv': None,
+        'whitenm': 1.0,
+        'pooling': 'gem',
+        'gemp': 3,
+        'use_gpu': True,
```

Comment:
Currently, whether to use GPU or not is decided in the parent module (e.g. `extract_features`) and set with `net.to(device)`. Can't we initialize the model on CPU and let the parent module move it to GPU later?

Reply:
When initializing on CPU it can't be moved to GPU later by simply using `Model(conf['model']).eval().to(device)`. I think `extract_features.py` may need some major changes for this and potentially other global extractors.